File: src/data-structure/shuffle.ts
const shuffle = (arr: number[]): number[] => {
  const ret = [...arr];
  const len = ret.length;
  let index = len - 1;
  while (index > 0) {
    const ranIndex = Math.floor(Math.random() * index);
    [ret[index], ret[ranIndex]] = [ret[ranIndex], ret[index]];
    index--;
  }
  return ret;
};

console.info(shuffle([1, 2, 3, 4, 5, 6]));

File: python/src/algorithm/shuffle.py
from random import randint


def shuffle(l):
    ret = [*l]
    length = len(ret)

    while length > 0:
        randi = randint(0, length - 1)
        ret[length-1], ret[randi] = ret[randi], ret[length-1]
        length -= 1

    return ret


l = [x for x in range(10)]
print(l)
print(shuffle(l))